File: Languages/C/primenumber.c
  #include<stdio.h>

int isPrime(int,int);

int main(){

    int num,prime;

    printf("Enter a positive number: ");
    scanf("%d",&num);

    prime = isPrime(num,num/2);

   if(prime==1)
        printf("%d is a prime number",num);
   else
      printf("%d is not a prime number",num);

   return 0;
}
// by recursion
int isPrime(int num,int i){

    if(i==1){
        return 1;
    }else{
       if(num%i==0)
         return 0;
       else
         isPrime(num,i-1);
    }
}

File: Languages/Java/prime-number.java
public class Prime {
    public static void main(String[] args) {
        int num = 29;
        boolean flag = true;
      	if(n==1)
			flag= false;

		if(n==2 || n==3)
			flag = true;

		if(n%2==0 || n%3==0)
			flag= false;

		for(int i=5; i*i<=n; i=i+6)
		{
			if(n % i == 0 || n % (i + 2) == 0)
				flag= false; 
		}

        if (!flag)
            System.out.println(num + " is a prime number.");
        else
            System.out.println(num + " is not a prime number.");
    }
}
